### PR TITLE
Active referrals counts increase only after the decrease active referrals count delay finished between 2 different (non-extended) mining sessions.

### PR DESCRIPTION
**Pull request by ice-myles**
Increment t1/t2 active counts are done by processors. Decrement is detected by miner. 

Miner handles batches of users, so the delay between `mine()` iteration start and decrementing user's `active_t1_referrals`/`active_t2_referrals` that was detected by `didReferralJustStopMining` function can create the situation,
when t1/t2 active count from previous mining session wasn't decremented, but has been already incremented for the new mining session. It happens only when user's session was finished, and then he started new session (not extended). As result t0 can have +1 obsolete `t1_active_count`/`t2_active_count` value.

To resolve this we need to return 409 duplicate error while t1/t2 referral count from previous session wasn't decreased. On production it can affect up to 9-10 seconds.

### CODE REVIEW

File: tokenomics/mining_sessions.go [hunk 1]
```diff
@@ -196,6 +204,7 @@ func (r *repository) newStartOrExtendMiningSession(old *StartOrExtendMiningSessi
 	resp.MiningSessionSoloEndedAt = time.New(now.Add(r.cfg.MiningSessionDuration.Max))
 	resp.MiningSessionSoloPreviouslyEndedAt = old.MiningSessionSoloEndedAt
 	resp.MiningSessionSoloDayOffLastAwardedAt = new(time.Time)
+	resp.ReferralsCountChangeGuardUpdatedAt = now
```
Comment (ice-ares) [inline tokenomics/mining_sessions.go]: Why cant u use MiningSessionSoloLastStartedAt?

Comment (ice-ares) [reply tokenomics/mining_sessions.go]: Oh, nvm, i see why

File: tokenomics/mining_sessions.go [hunk 2]
```diff
@@ -76,6 +78,12 @@ func (r *repository) StartNewMiningSession( //nolint:funlen,gocognit // A lot of
 		(now.Sub(*old[0].MiningSessionSoloLastStartedAt.Time)/r.cfg.MiningSessionDuration.Min)%2 == 0 {
 		return ErrDuplicate
 	}
+	if !old[0].MiningSessionSoloEndedAt.IsNil() &&
+		!old[0].ReferralsCountChangeGuardUpdatedAt.IsNil() &&
+		now.After(*old[0].MiningSessionSoloEndedAt.Time) &&
+		old[0].MiningSessionSoloStartedAt.Equal(*old[0].ReferralsCountChangeGuardUpdatedAt.Time) {
```
Comment (ice-cronus) [inline tokenomics/mining_sessions.go:84]: isn't it `old.[0].ReferralsCountChangeGuardUpdatedAt.After(*old[0].MiningSessionSoloEndedAt.Time)`?
As you updating value to now in miner, if session was processed all referralsChange its ReferralsCountChangeGuardUpdatedAt become greater than session end time, but equal may not be valid here if session takes more than 1 miner iteration isn't it? Cuz they're equal only after session creation.

Comment (ice-myles) [reply tokenomics/mining_sessions.go:84]: `MiningSessionSoloStartedAt` is not changed if session is extended.
And this `ReferralsCountChangeGuardUpdatedAt` is changed only when `didReferralJustStopMining` returns non-nil value. `didReferralJustStopMining` notifies that user stopped to mine.

Also `now.After(*old[0].MiningSessionSoloEndedAt.Time)` check at mining_session.go checks the only case when user tries to start new session (to filter session extending functionality).

**Pull request merged**